ci/genericize-stream.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/python3
+
+import json
+import sys
+
+def walk(item, cb_obj=None, cb_list=None, cb_str=None):
+    '''Recursively descend a JSON dict, calling callbacks for each type.
+    cb_str() must return the replacement string.'''
+    if isinstance(item, dict):
+        ret = {}
+        for k, v in item.items():
+            ret[k] = walk(v, cb_obj, cb_list, cb_str)
+        if cb_obj:
+            cb_obj(ret)
+        return ret
+    elif isinstance(item, list):
+        ret = []
+        for v in item:
+            ret.append(walk(v, cb_obj, cb_list, cb_str))
+        if cb_list:
+            cb_list(ret)
+        return ret
+    elif isinstance(item, str):
+        if cb_str:
+            item = cb_str(item)
+        return item
+    else:
+        return item
+
+
+def replace_string(tree, old, new):
+    '''Replace substring old with new throughout tree.'''
+    return walk(tree, cb_str=lambda s: s.replace(old, new))
+
+
+def replace_key(tree, key, new):
+    '''Replace the value of key with new throughout tree.'''
+    def cb_obj(o):
+        if key in o:
+            o[key] = new
```

Comment:
Should we verify here that the type of `o[key]` and `new` are actually still the same?

Reply:
Eh, I don't think it matters. `new` is a placeholder value, and `check-diff` is doing a textual rather than semantic comparison of the output.